### ISSUE TITLE
Enforce `.replaceAll()` over `.split().join()` in `prefer-string-replace-all` rule

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1441]
## Fail

```js
foo.split('a').join('b');
```

## Pass

```js
foo.replaceAll('a', 'b');
```

**Comment by sindresorhus** [comment]
This is accepted.

**Comment by papb** [comment]
> ```js
> // Not sure about this, maybe add `g` flag, result the same?
> foo.split(/a+/).join('b');
> ```

It's important to remember that calling `split` with a regular expression has some gotchas, such as if the regex has a capturing group:

```js
'a_b_c'.split(/_/);
//=> [ 'a', 'b', 'c' ]
'a_b_c'.split(/(_)/);
//=> [ 'a', '_', 'b', '_', 'c' ]
```

**Comment by fisker** [comment]
The second argument also little different https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_string_as_the_replacement


```console
'1_2_3'.split('_').join('$&')
//=> '1$&2$&3'
'1_2_3'.replaceAll('_', '$&')
//=> '1_2_3'
'1_2_3'.split('_').join('$$')
//=> '1$$2$$3'
'1_2_3'.replaceAll('_', '$$')
//=> '1$2$3'
```

**Comment by andersk** [comment]
The fix can double each `$` to `$$` in the replacement argument if it’s known to be a string literal (e.g. `.split('_').join('$$')` → `.replaceAll('_', '$$$$')`), and otherwise use a function (e.g. `.split('_').join(s)` → `.replaceAll('_', () => s)`).
